internal/campaigns/executor.go

```diff
@@ -144,15 +144,19 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 		if result, err = x.cache.Get(ctx, cacheKey); err != nil {
 			err = errors.Wrapf(err, "checking cache for %q", task.Repository.Name)
 			return
-		} else if result != nil {
+		} else if result != nil && len(result.Commits) == 1 {
+			// Build the changeset spec.
```

Comment:
Can you expand this out a little to note _why_ we're rebuilding the spec from the cached diff, please? (It took me a couple of reads to realise that it's because the changeset template may have changed, which might affect the "commit" that gets embedded in the changeset spec.)

Reply:
Hopefully my new comment is more clear. Thanks!

Reply:
It is. Thanks!